### ISSUE TITLE
Bracket pair colorization - skip or detect Vue and Liquid template tags

### ISSUE THREAD

**Issue by lorand-horvath** [microsoft/vscode#132240]
The new bracket pair colorization feature in VSCode 1.60 uses different colors for the inner and outer curly brackets in certain templating languages such as Shopify Liquid or Vue's string interpolation in templates.

Example in Liquid and Vue: {{ code }} should have both pair of brackets of the same color, not a yellow and a magenta one - it's tough on the eyes. Preferably, both inner and outer pairs should be either yellow or white or gray.

Also, in Liquid, the control flow tags {% code %} have the curly brackets yellow and the percentage signs as gray - it also looks weird. {% and %} should both be of the same color (yellow, white or gray).

It would be preferable to have an option added to "Bracket pair colorization" to be able to exclude certain delimiters from being colorized. For example: `excludedDelimiters: ["{{", "}}", "{%", "%}"]` or to be able to choose their specific color.

**Comment by lorand-horvath** [comment]
Additional issue: Bracket pairs should not be colorized in .txt files or any other non-programming language related files. I guess that should have been obvious from the very start, before pushing out the native pair brackets colorization support at all.

**Comment by hediet** [comment]
> to be able to choose their specific color.

This is already possible!
A language can define [bracket pairs to colorize](https://github.com/microsoft/vscode/blob/997228d528d687fd17cbf3ba117a0d4f668c1393/extensions/javascript/tags-language-configuration.json#L11).

> Additional issue: Bracket pairs should not be colorized in .txt files or any other non-programming language related files. I guess that should have been obvious from the very start, before pushing out the native pair brackets colorization support at all.

Please open a separate issue for that!

**Comment by lorand-horvath** [comment]
@hediet Thanks for your response. The bracket paris to colorize options are not in the documentation yet, that is why I haven't found a solution to this.
To have {{ and }} of the same color, do I have to actually change the file here
c:\Program Files\Microsoft VS Code\resources\app\extensions\javascript\tags-language-configuration.json
or is there a less messy solution to this? I would like to have {{ }} {% %} colored in the same color in Vue and Liquid files, but not in JS files in general.

**Comment by lorand-horvath** [comment]
@hediet I've managed to fix this by manually updating the HTML (not JS) configuration file here:
`c:\Program Files\Microsoft VS Code\resources\app\extensions\html\language-configuration.json`
and added the bracket pairs to colorize with the same color like so:
`"colorizedBracketPairs":[["{","}"],["(",")"],["{{","}}"],["{%","%}"]]`

Can we have this added to the html/language-configuration.json by default, please? Or at least have an option to manually add them in the VSCode Settings interface under Extensions/HTML ?

**Comment by hediet** [comment]
> Can we have this added to the html/language-configuration.json by default, please?

Yes. We will do that for all native language configs, but it also needs to be done for community language configs.

If you like, you can submit your change as PR! Otherwise I'll do the change in the next couple of days.
Thanks!

**Comment by lorand-horvath** [comment]
@hediet I would happily submit a PR, if I would know how to handle specific files, such as `.txt` or `.vue`. Is this what you mean by "community language configs"? Sorry, but I'm a bit new to VSCode open source coding.

**Comment by thernstig** [comment]
This problem also exists in files using the Helm templating language, i.e. the `helm-template` identifier.
